internal/ingest/ingest.go

```diff
@@ -136,11 +136,11 @@ type Ingester struct {
 
 	// Multihash minimum length
 	minKeyLen int
-	// chunkConcurrency is accessed through SetEntriesChunkConcurrency() and
-	// entriesChunkConcurrency(). It tells the Ingester to handle writing entry chunks
-	// asynchronously using this main additional goroutines per
-	// publisher/worker.
-	chunkConcurrency int32
+
+	chunkConcurrency int
+	chunkWorkQueue   chan func()
+	// chunkConcurrencyRWMute protects access to chunkConcurrency and chunkWorkQueue.
+	chunkConcurrencyRWMutex sync.RWMutex
```

Comment:
this mutex is only in place to be able to set the concurrency?

Reply:
one q: is there any chance that processing of a chunk can generate new work? are there issues with the changing of chunk size holding the write lock while finishing the processing queue?

Reply:
Yes. If it were not runtime configurable, then it would be unnecessary.